File: BE/gumid207/src/main/java/com/ssafy/gumid207/dto/UserDto.java
package com.ssafy.gumid207.dto;

import java.io.Serializable;
import java.time.LocalDateTime;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.ssafy.gumid207.entity.User;

import io.swagger.annotations.ApiParam;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 직렬화 기능을 가진 User클래스
 */
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserDto implements Serializable {

	private Long userSeq;

	@ApiParam(value = "유저 이름")
	private String userNickName;

	@ApiParam(value = "유저 포인트")
	private Integer userPoint;

	@ApiParam(value = "유저 아이디")
	private String userId;

	@ApiParam(value = "유저 비밀번호")
	private String userPass;

	@ApiParam(value = "유저 이메일")
	private String userEmail;

	@ApiParam(value = "유저 생년")
	private Integer userBirthday;

	@ApiParam(value = "유저 성별")
	private String userGender;

	@ApiParam(value = "유저 fcm 토큰")
	private String userFcmToken;

	@ApiParam(value = "등록 시간. (yyyy-MM-dd 00:00:00)")
	@DateTimeFormat(pattern = "yyyy-MM-dd HH:mm:ss")
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
	private LocalDateTime userRegTime;

	public static UserDto of(User user) {
		if (user == null) {
			return null;
		}
		return new UserDtoBuilder() //
				.userSeq(user.getUserSeq()) //
				.userPoint(user.getPoint()) //
				.userNickName(user.getNickName()) //
				.userId(user.getId()) //
				.userPass(user.getPass()) //
				.userEmail(user.getEmail()) //
				.userBirthday(user.getBirthday()) //
				.userGender(user.getGender()) //
				.userFcmToken(user.getFcmToken()) //
				.userRegTime(user.getRegTime()) //
				.build();
	}

}

File: AOS/SongForYou/app/src/main/java/com/hanyeop/songforyou/model/dto/UserDto.kt
package com.hanyeop.songforyou.model.dto

import com.google.gson.annotations.SerializedName

data class UserDto(
    //@SerializedName("userSeq") val userSeq: Int,
    //@SerializedName("userVoiceFileSeq") val userVoiceFileSeq: Int,
//    @SerializedName("userBirthday") val userBirthday: Int,                      // 출생년도
//    @SerializedName("userEmail") val userEmail: String,                         // 이메일
//    @SerializedName("userGender") val userGender: String,                       // 성별
//    @SerializedName("userId") val userId: String,                               // id
//    @SerializedName("userNickname") val userNickname: String,                   // 별명
//    @SerializedName("userPass") val userPass: String,                           // 비밀번호
//    @SerializedName("userProfileImgSeq") val userProfileImgSeq: Int,            // 프로필
//    @SerializedName("userRegTime") val userRegTime: String,                     // 등록시간

    @SerializedName("birthday") val userBirthday: Int,                      // 출생년도
    @SerializedName("email") val userEmail: String,                         // 이메일
    @SerializedName("gender") val userGender: String,                       // 성별
    @SerializedName("id") val userId: String,                               // id
    @SerializedName("nickName") val userNickname: String,                   // 별명
    @SerializedName("pass") val userPass: String,                           // 비밀번호
//    @SerializedName("userProfileImgSeq") val userProfileImgSeq: Int,            // 프로필
//    @SerializedName("userRegTime") val userRegTime: String,                     // 등록시간
)
// id, 프로필 이미지 번호 왜 필요한지 물어보기